src/APNSClient.php

```diff
@@ -95,56 +72,15 @@ private function enqueueRequest( APNSRequest $request ): void {
 	 */
 	private function sendQueuedRequests(): array {
 
-		$responses = [];
-
-		do {
-			$status = curl_multi_exec( $this->curl_handle, $running_operation_count );
-
-			while ( true ) {
-
-				$info = curl_multi_info_read( $this->curl_handle );
-
-				if ( ! $info ) {
-					break;
-				}
-
-				if ( $info['result'] !== CURLE_OK ) {
-					throw new Exception( 'Request failed: ' . $info['result'] );
-				}
+		$network_responses = $this->network_service->sendQueuedRequests();
 
-				if ( ! is_null( $info['handle'] ) ) {
-					$handle = $info['handle'];
-					$responses[] = $this->processResponse( $handle );
+		$apns_responses = [];
 
-					curl_multi_remove_handle( $this->curl_handle, $handle );
-					curl_close( $handle );
-				}
-			}
-		} while ( $running_operation_count > 0 && $status === CURLM_OK );
-
-		if ( $status !== CURLM_OK ) {
-			throw new Exception( 'Unable to continue sending – ' . curl_multi_strerror( $status ) );
+		foreach ( $network_responses as $response ) {
+			$apns_responses[] = $response;
 		}
 
-		return $responses;
-	}
-
-	/**
-	 * Parse a completed CURL handle into an APNSResponse object
-	 *
-	 * @param resource $handle
-	 */
-	private function processResponse( $handle ): APNSResponse {
```

Comment:
This is not required anymore now that the network layer returns an `APNSResponse`.